.travis.yml

```diff
@@ -55,57 +48,46 @@ matrix:
-  # Create WordPress database.
-  - mysql -e 'CREATE DATABASE wordpress_test;' -uroot
-  # Install CodeSniffer for WordPress Coding Standards checks.
-  - git clone https://github.com/squizlabs/PHP_CodeSniffer.git php-codesniffer
-  # Install WordPress Coding Standards.
-  - git clone https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards.git wordpress-coding-standards
-  # Hop into CodeSniffer directory.
-  - cd php-codesniffer
-  # Set install path for WordPress Coding Standards.
-  # @link https://github.com/squizlabs/PHP_CodeSniffer/blob/4237c2fc98cc838730b76ee9cee316f99286a2a7/CodeSniffer.php#L1941
-  - scripts/phpcs --config-set installed_paths ../wordpress-coding-standards
-  # Hop into themes directory.
-  - cd $theme_dir
-  # After CodeSniffer install you should refresh your path.
-  - phpenv rehash
+    - export PHPCS_DIR=/tmp/phpcs
+    - export WPCS_DIR=/tmp/wpcs
+    # Install CodeSniffer for WordPress Coding Standards checks.
+    - if [[ "$SNIFF" == "1" ]]; then git clone -b master --depth 1 https://github.com/squizlabs/PHP_CodeSniffer.git $PHPCS_DIR; fi
+    # Install WordPress Coding Standards.
+    - if [[ "$SNIFF" == "1" ]]; then git clone -b master --depth 1 https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards.git $WPCS_DIR; fi
+    # Set install path for WordPress Coding Standards.
+    # @link https://github.com/squizlabs/PHP_CodeSniffer/blob/4237c2fc98cc838730b76ee9cee316f99286a2a7/CodeSniffer.php#L1941
+    - if [[ "$SNIFF" == "1" ]]; then $PHPCS_DIR/scripts/phpcs --config-set installed_paths $WPCS_DIR; fi
+    # After CodeSniffer install you should refresh your path.
+    - if [[ "$SNIFF" == "1" ]]; then phpenv rehash; fi
+    # Install JSCS: JavaScript Code Style checker
+    # @link http://jscs.info/
+    - if [[ "$SNIFF" == "1" ]]; then npm install -g jscs; fi
+    # Install JSHint, a JavaScript Code Quality Tool
+    # @link http://jshint.com/docs/
+    - if [[ "$SNIFF" == "1" ]]; then npm install -g jshint; fi
 
 # Run test script commands.
 # Default is specific to project language.
 # All commands must exit with code 0 on success. Anything else is considered failure.
 script:
-  # Search for PHP syntax errors.
-  - find . \( -name '*.php' \) -exec php -lf {} \;
-  # WordPress Coding Standards
-  # @link https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
-  # @link http://pear.php.net/package/PHP_CodeSniffer/
-  # -p flag: Show progress of the run.
-  # -s flag: Show sniff codes in all reports.
-  # -v flag: Print verbose output.
-  # -n flag: Do not print warnings. (shortcut for --warning-severity=0)
-  # --standard: Use WordPress as the standard.
-  # --extensions: Only sniff PHP files.
-  - $WP_DEVELOP_DIR/php-codesniffer/scripts/phpcs -p -s -v -n . --standard=./codesniffer.ruleset.xml --extensions=php
+    # Search for PHP syntax errors.
+    - find -L . -name '*.php' -print0 | xargs -0 -n 1 -P 4 php -l
```

Comment:
> Is this the only line that'll get run against the matrix?

True - In reality, we could even remove some PHP versions and only run this lint against PHP 5.2, 5.6, 7.0 and HHVM. I don't think the versions in between will throw different parse errors which aren't caught by running just those four.

> Looks like everything else is relegated to a single sniff (which is fine).

Yes, the results of everything else won't change across PHP versions.